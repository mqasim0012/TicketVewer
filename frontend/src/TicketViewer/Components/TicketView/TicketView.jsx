import { React, useEffect, useState } from "react";

import getTickets from "../../getTickets";
import Ticket from "../Ticket";
import Error from "../Error";

// import LoadingButton from '@mui/lab/LoadingButton';

const TicketView = ({props}) => {
    const [loading, setLoading] = useState(false);
    const [content, setContent] = useState([]);
    const [error, setError] = useState();

    useEffect(() => {
        console.log("Content changed:")
        console.log(content)
    }, [content])

    useEffect(() => {
        setLoading(true);
        getTickets()
            .then(res => {
                console.log("Successful response communicated to ticketViewer")
                console.log(res.data.requests)
                setContent(res.data.requests);
            })
            .catch(err => {
                console.log("Error communicated to ticketViewer")
                setError(err)
                console.log(err)
            })
            .finally(() => {
                setLoading(false)
            })
    }, []);

    return (
        <div>
            <h1>Ticket view</h1><br/>
            {!loading && !error && <p>Click on any ticket to view</p>}
            {loading &&
                "Loading..."
            }
            {
                error && <Error error={error.error} description={error.error_description} />
            }
            {
                content.map(ticket => (
                    <Ticket description={ticket.description} />
                ))
            }
        </div>
    )
};

export default TicketView;