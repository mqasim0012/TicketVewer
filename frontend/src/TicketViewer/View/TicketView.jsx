import { useEffect } from "react";
import getUser from "../api";

const TicketView = ({props}) => {
    // const [content, setContent] = useState(20);

    useEffect(() => {
        getUser();
    });

    return (
        <div>
            ticket view <br/>

        </div>
    )
};

export default TicketView;