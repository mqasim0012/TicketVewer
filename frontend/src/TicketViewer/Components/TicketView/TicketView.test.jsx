import React from 'react'
import renderer from 'react-test-renderer'
import { render, screen, fireEvent } from '@testing-library/react'

import '@testing-library/jest-dom';
import TicketView from '.'

describe("Testing ticket viewer dashboard", () => {
    test("Test renders correctly", ()=>{

        const sampleTicket = renderer.create(
            <TicketView
            />
        )

        const tree = sampleTicket.toJSON();
        expect(tree).toMatchSnapshot();
    })

    // test("Prev and next buttons", () => {

    //     render(<TicketView />)
    // })
})