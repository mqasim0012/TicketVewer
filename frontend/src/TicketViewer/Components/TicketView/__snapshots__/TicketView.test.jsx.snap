// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing ticket viewer dashboard Test renders correctly 1`] = `
<div>
  <h1>
    Ticket view
  </h1>
  <p>
    Click on any ticket to view
  </p>
  <p>
    Showing 
    1
    -
    0
     of 
    0
     tickets
  </p>
</div>
`;
